src/catalog/src/builtin.rs

```diff
@@ -2162,8 +2180,15 @@ pub static MZ_COMPUTE_DEPENDENCIES: LazyLock<BuiltinSource> = LazyLock::new(|| B
         .with_column("object_id", ScalarType::String.nullable(false))
         .with_column("dependency_id", ScalarType::String.nullable(false))
         .finish(),
+    column_comments: Some(vec![
+        "The ID of a compute object. Corresponds to `mz_catalog.mz_indexes.id`, \
+        `mz_catalog.mz_materialized_views.id`, or `mz_internal.mz_subscriptions`",
+        "The ID of a compute dependency. Corresponds to `mz_catalog.mz_indexes.id`, \
+        `mz_catalog.mz_materialized_views.id`, `mz_catalog.mz_sources.id`, or `mz_catalog.mz_tables.id`",
+    ]),
```

Comment:
I would prefer an API that takes a `(column_index, comment)` or `(column_name, comment)` pair, this way you don't have to leave a comment on all columns. But as-is this is a net win, so not blocking!